src/DataProtection/Cryptography.Internal/src/SafeHandles/BCryptAlgorithmHandle.cs

```diff
@@ -65,25 +65,22 @@ public BCryptKeyHandle GenerateSymmetricKey(byte* pbSecret, uint cbSecret)
         /// </summary>
         public string GetAlgorithmName()
         {
+            const int StackAllocCharSize = 128;
```

Comment:
There can't be longer algorithm names?
I gues no, but ... want to double-check.

Reply:
I'd be _extremely_ surprised if that were the case. So what you had proposed should be good.

If you wanted to be defensive you could always `string.Create` instead of stackalloc, but overflowing this buffer is never going to happen in practice. (This method isn't even called outside unit tests TBH.)